### DISCUSSION TITLE
Setup for Barebones ie Ubuntu Debian

### DISCUSSION THREAD

**Discussion by DevilsDesigns** [photoview/photoview#186] [Q&A and Troubleshooting]
is there a way to use this on a barebones machine ie. Ubuntu 20.04, Debina 10 without a docker install? If so can you Specify.

**Comment by sh4nks** [comment]
Set it up like it's specified in the [Development Environment](https://github.com/photoview/photoview#setup-development-environment) setup instructions. Change the ``/api/.env`` accordingly (no debug mode, ui endpoint). Build the ui in ``/ui/`` with ``npm run build`` and make sure to point the API_ENDPOINT in ``/ui/.env/`` to the correct API address. The files in ``/ui/dist/`` can be hosted with apache or nginx (= this is the UI_ENDPOINT then).

**Comment by viktorstrate** [comment]
I wan't to write a propper guide explaining how to do this, I just haven't gotten around to do so yet.

I have written a guide a while ago that you can use, everything should be the same, except some of the environment variables has been changed a little with version 2. See https://github.com/photoview/photoview/issues/70#issuecomment-676255190

**Comment by gdepre** [comment]
did you manage to make it work ?

I've got problems with Go dependencies, with Package libheif. So I can't run it, & docker is not trusted in my company ^^

**Reply by viktorstrate** [reply]
Yes you'll need libheif v1.11 (older versions don't work) to build and run Photoview now, this dependency is relativeley new and I haven't updated the "Manual Install guide" from the docs yet.

You can download a package for Ubuntu using this PPA:
https://launchpad.net/~strukturag/+archive/ubuntu/libheif
